tests/test_engine.py

```diff
@@ -252,31 +253,36 @@ def test_save_load_checkpoint(engine):
     """
     # Path of the input directory
     ckpt_path = config.save["engine_checkpoint"]
-
-    # Get initial weights
-    init_weights =[p.clone() for p in engine.network.parameters()]
+    # Path of reference checkpoint
+    ref_path =  config.save["reference_checkpoint"]
 
     assert os.path.exists(ckpt_path)
+    assert os.path.exists(ref_path)
 
-    # Save checkpoint
-    engine.save_checkpoint(ckpt_path)
-
+    # Test for saving checkpoint
```

Comment:
you are not even using the save function here, how are you testing it?
The logic of this function should be something like this (pseudocode):

\# Test load
torch_loaded_weights = torch.load(reference_load_cp)
engine.load(reference_load_cp)
engine_loaded_weights = engine.network.get_weights()
assert torch_loaded_weights == engine_loaded_weights # if true, then load function is working properly.

\# Test save
initial_weights = engine.network.get_weights()
assert not exists(saved_cp)
engine.save(saved_cp)
assert exists(saved_cp) #make sure a cp is saved
engine.load(saved_cp) # at this point, we know that load is functioning properly
saved_weights = engine.network.get_weights()
assert saved_weights == initial_weights #if they are not equal, and load is functioning properly, then save is not
delete(saved_cp)